exercise-solutions/tcp-server-exercises/src/bin/echo-log-arc-mutex.rs

```diff
@@ -0,0 +1,37 @@
+use std::{
+    io::{self, BufRead as _, BufReader, BufWriter, Write as _},
+    net::{TcpListener, TcpStream},
+    sync::{Arc, Mutex, MutexGuard},
+    thread,
+};
+
+fn handle_client(stream: TcpStream, log: &Mutex<Vec<usize>>) -> Result<(), io::Error> {
+    let reader = BufReader::new(&stream);
+    let mut writer = BufWriter::new(&stream);
+    for line in reader.lines() {
+        let line = line?;
+        // the code block here forces the MutexGuard drop to unlock the mutex
+        {
+            let mut log: MutexGuard<'_, _> = log.lock().unwrap();
+            log.push(line.len());
+        }
+        writer.write_all(line.as_bytes())?;
+        writer.write_all(b"\n")?;
+        writer.flush()?;
+    }
+    Ok(())
+}
+
+fn main() -> Result<(), io::Error> {
+    let log = Arc::new(Mutex::new(vec![]));
+
+    let listener = TcpListener::bind("127.0.0.1:7878")?;
+    for stream in listener.incoming() {
+        let stream = stream?;
```

Comment:
This will exit the program if a client fails to connect. Perhaps `let Ok(stream) = stream_result else { println!("Bad connection"); continue; }`?